lib/streamlit/hashing_py3.py

```diff
@@ -32,14 +69,12 @@ def set_tos(t):
             refs.append(tos)
         tos = t
 
-    # Our goal is to find referenced objects. The problem is that co_names
-    # does not have full qualified names in it. So if you access `foo.bar`,
-    # co_names has `foo` and `bar` in it but it doesn't tell us that the
-    # code reads `bar` of `foo`. We are going over the bytecode to resolve
-    # from which object an attribute is requested.
-    # Read more about bytecode at https://docs.python.org/3/library/dis.html
+    def handle_operation(op):
```

Comment:
Would be cleaner if this function just took `lineno` and `tos` as arguments and returned `lineno, tos`. It's always best to have side-effect-free functions, as they're easier to understand.

Then you can also move `handle_operation` out of here and into the top level scope, which makes it clear that it doesn't use a closure.

Reply:
Seems like there needs to be some additional changes for this to work, as it's breaking the current logic in the function by only doing what you've described. 

I got rid of the `handle_operation` function and moved that code back inside the try block, where it was before my changes.